crates/ty_server/tests/e2e/main.rs

```diff
@@ -417,7 +468,54 @@ impl TestServer {
     ///
     /// This method will remove the notification from the internal data structure, so it should
     /// only be called if the notification is expected to be sent by the server.
-    pub(crate) fn await_notification<N: Notification>(&mut self) -> Result<N::Params> {
+    ///
+    /// # Panics
+    ///
+    /// If the server doesn't send the notification within the default timeout or
+    /// the notification failed to deserialize. Use [`Self::try_await_notification`] for
+    /// a panic-free alternative.
+    #[track_caller]
+    pub(crate) fn await_notification<N: Notification>(&mut self) -> N::Params {
+        match self.try_await_notification::<N>(None) {
+            Ok(params) => params,
+            Err(err) => match err {
+                TestServerError::RecvTimeoutError(RecvTimeoutError::Disconnected) => {
```

Comment:
Same here although this is a bit different but the response error variant could be moved first to use `unreachable` and then a catch-all `err` for the remaining error.